### ISSUE TITLE
TOML-to-Header parser fails on edge-case

### ISSUE THREAD

**Issue by matthiasbeyer** [matthiasbeyer/imag#163]
When calling:

```
./target/debug/imag-store --debug --rtp /tmp/ create -p /test entry -c content -h header=1
```

(from https://github.com/matthiasbeyer/imag/commit/57a12e1a0e90b6c10759d6ff93b37be35c41a7f1)

It generates the following file:

```
---header = "1"

[imag]
links = []
version = "0.1.0"
---
content 
```

Which, of course, fails to parse on `retrieve()`.

**Comment by TheNeikos** [comment]
This is not an error in the store, unless you want us to check the header on every write? Which could be trivial, but basically all the values in the first map have to be tables.

**Comment by matthiasbeyer** [comment]
Right, this is in the doc already. I'm not sure where to do the sanity check, though. I guess in the store would be the right place.

**Comment by matthiasbeyer** [comment]
@TheNeikos When will you start working on this one?

**Comment by TheNeikos** [comment]
Now

**Comment by matthiasbeyer** [comment]
:+1:

**Comment by matthiasbeyer** [comment]
Issue was not fixed by #185 , though #185 is not invalid.

**Comment by matthiasbeyer** [comment]
Funny enough:

The example from above no longer produces an invalid header.

The error message which is printed to the user is still not helpful: There is none. The exit code is wrong as well (`0`, should be `!= 0`). I will open a new issue for this.
